src/features/auth/components/AuthModal.tsx

```diff
@@ -0,0 +1,84 @@
+import { GithubAuthProvider, GoogleAuthProvider, signInWithPopup } from 'firebase/auth'
+import { FaGithub, FaGoogle } from 'react-icons/fa'
+import ReactModal from 'react-modal'
+import toast from 'react-simple-toasts'
+import { auth, githubProvider, googleProvider } from '../api/Config'
+
+type AuthModalProps = {
+  showAuth: boolean
+  setShowAuth: (show: boolean) => void
+}
+
+export const AuthModal = ({ showAuth, setShowAuth }: AuthModalProps) => {
+  const signInWithGoogle = () => {
+    signInWithPopup(auth, googleProvider)
+      .then((result) => {
+        const credential = GoogleAuthProvider.credentialFromResult(result)
+        // TODO save this token in user settings maybe!
+        const token = credential.accessToken
+        // The signed-in user info.
+        const user = result.user
```

Comment:
Create a user store to save the user token and details, so the data persists.
like https://github.com/medyo/hackertab.dev/blob/master/src/stores/bookmarks.ts